src/JsonaTypes.ts

```diff
@@ -45,26 +45,26 @@ export interface IJsonaDeserializer extends IJsonaModelBuilder {
 }
 
 export interface IJsonDeserializerConstructor {
-    new(propertiesMapper: IJsonPropertiesMapper, deserializeCache: IDeserializeCache, options);
+    new(propertiesMapper: IJsonPropertiesMapper, deserializeCache: IDeserializeCache, options?: TDeserializeOptions): IJsonaDeserializer;
 }
 
 export interface IModelsSerializer {
-    setPropertiesMapper(propertiesMapper: IModelPropertiesMapper);
-    setStuff(stuff);
-    setIncludeNames(includeNames: TJsonaDenormalizedIncludeNames | TJsonaNormalizedIncludeNamesTree);
+    setPropertiesMapper(propertiesMapper: IModelPropertiesMapper): void;
+    setStuff(stuff:  TJsonaModel | TJsonaModel[]): void;
+    setIncludeNames(includeNames: TJsonaDenormalizedIncludeNames | TJsonaNormalizedIncludeNamesTree): void;
     build(): TJsonApiBody;
     buildDataByModel(model: TJsonaModel | null): TJsonApiData;
-    buildRelationshipsByModel(model: TJsonaModel);
+    buildRelationshipsByModel(model: TJsonaModel): TJsonaRelationshipDataItem[] | TJsonaModel | TJsonaRelationshipBuild;
```

Comment:
It should be
` buildRelationshipsByModel(model: TJsonaModel): TJsonaRelationships;`

Reply:
Thank you for the review!

`TJsonaRelationships` is an object defined as `Record<string, ...>` with a type that the `buildRelationshipsByModel` method returns. I think the best solution would be to define a type `TJsonaRelationship` as `TJsonaRelationshipBuild | TJsonaModel | Array<TJsonaModel>` and reuse it in both this method and `TJsonaRelationships` as a prop type.

Reply:
Hi, 
1. I was wrong about offering `TJsonaRelationships` as a return type of `buildRelationshipsByModel`

2. Models serializer cannot produce TJsonaModel because it should create json from the models.

3. `buildRelationshipsByModel` should use 
```
{
  [relationName: string]: {
    data: TResourceIdObj
  }
}
```
as a return type